python/setup.py

```diff
@@ -28,7 +28,7 @@
 else:
     from setuptools import setup
     from setuptools.extension import Extension
-    kwargs = {'install_requires': ['numpy', 'requests', 'graphviz'], 'zip_safe': False}
+    kwargs = {'install_requires': ['numpy<=1.13.3', 'requests==2.18.4', 'graphviz==0.8.1'], 'zip_safe': False}
```

Comment:
==1.13.3

Reply:
Jonathan - I was thinking of large majority of users who has 1.8.2 (lower than 1.13.3) version of Numpy. Which we have tested works well with MXNet. So I thought of capping to the major version that is latest and tested rather than forcing users to upgrade.

Reply:
I like the idea of making the dependency versions deterministic. However, we should be careful and diligently test and upgrade the versions with our releases, newer versions of the packages might release performance fixes, security fixes, etc., Lets add testing and upgrading dependent package version to MXNet Release checklist.  
does the pip package builder use the setup.py from here?

Reply:
While the dependency string "numpy<=1.13.3" is more flexible in supporting a variety of numpy versions, the drawback/tradeoff is that if a particular lower version of numpy is NOT tested then the combinations of numpy versions may make it hard to test and support. I think that the default version should be what the MXNet CI tests and then users can change the script to make it flexible if required.

Reply:
numpy 1.8.2 is more than 3 years old. why do we think a large majority of users have that version?

we can also specify a range of versions. e.g. numpy >= 1.8, < 1.14, but as @nswamy says, the range should only include versions we have tested with the release.

another approach to this issue would be to investigate why our setup script is accepting pre-release versions of dependencies and fix that. pip installs correctly ignore 1.14.0rc1 (without --pre), but our setup script does not.